### PR TITLE
feat: add eventbridge provider

### PR DESCRIPTION
**Pull request by rodrigomafrarios**
- [x] unit tests
- [x] add AWS EventBridge provider

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,12 +12,16 @@
     "clean": "yarn tsc -b --clean",
     "release": "yarn build && yarn changeset publish"
   },
-  "dependencies": {},
+  "dependencies": {
+    "@model-ts/core": "^0.1.2",
+    "tslib": "^2.3.1"
```
Comment (finkef) [inline package.json:17]: I think these are not required here

File: packages/eventbridge/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "tslib": "^2.3.0",
+    "typescript": "~4.3.5",
+    "uuid": "^8.3.2"
+  },
+  "dependencies": {
+    "dataloader": "^2.0.0",
+    "io-ts-reporters": "^1.2.2",
+    "retry-ts": "^0.1.2",
+    "snapshot-diff": "^0.8.1"
+  }
```
Comment (finkef) [inline packages/eventbridge/package.json:40]: Are these actually required?

File: packages/eventbridge/src/errors.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      message: this.message,
+    }
+  }
+
+  toHttpError() {
+    return {
+      statusCode: 500,
+      body: this.message,
+    }
+  }
```
Comment (finkef) [inline packages/eventbridge/src/errors.ts:22]: I think this made it over from the internal codebase by mistake.

File: packages/eventbridge/src/provider.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            EventBusName: process.env.EVENT_BUS,
+            Source: this.source,
+            DetailType: this.detailType,
+            Detail: this.encode(),
+          }]
+        }).promise()
+
+        const failed = result.FailedEntryCount || !result.Entries
+        if (failed) {
+          throw new InternalServerError()
```
Comment (finkef) [inline packages/eventbridge/src/provider.ts:28]: Does it make sense to throw a more specialized error, maybe something along the lines of `PublishError` or even more descriptive based on the error reason?
Maybe also take a look here for inspiration: https://github.com/finkef/model-ts/blob/main/packages/dynamodb/src/errors.ts

**Review by finkef — CHANGES_REQUESTED** [review]

File: packages/eventbridge/src/provider.ts (hunk 4 above)
Comment (rodrigomafrarios) [reply packages/eventbridge/src/provider.ts]: @finkef done. check it out

File: packages/eventbridge/src/errors.ts (hunk 3 above)
Comment (rodrigomafrarios) [reply packages/eventbridge/src/errors.ts]: you're right. removed.

File: packages/eventbridge/package.json (hunk 2 above)
Comment (rodrigomafrarios) [reply packages/eventbridge/package.json]: no, removing

File: package.json (hunk 1 above)
Comment (rodrigomafrarios) [reply package.json:17]: you're right. removing

**Comment by finkef** [comment]
LGTM! 🎉

**Pull request merged**